### PR TITLE
issue-91: Fix munchify for tuples of lists

### LINKED ISSUE #91: munchify broken?
Here is a minimal working example, which is broken.   Is this already fixed?

`munchify((1,2,[{'A': 'B'}])) 
Result: (1, 2, [Munch({'A': 'B'}), Munch({'A': 'B'})])
Expected:  (1, 2, [Munch({'A': 'B'})])`

### PR DESCRIPTION
**Pull request by d1618033**
fixes #91

### CODE REVIEW

File: tests/test_munch.py [hunk 1]
```diff
@@ -415,6 +415,7 @@ def test_munchify_cycle():
     m = munchify(x)
     assert m.id == "x"
     assert m.y[0] == "y"
+    assert m.y[1] == m
```
Comment (NoamNol) [inline tests/test_munch.py]: Not necessary if we already have `assert m.y[1] is m`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,11 +40,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
-    # Install the testing dependencies from setup.cfg, including pylint
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
```
Comment (NoamNol) [inline .github/workflows/ci.yml:46]: I think we want to test all supported versions

File: munch/__init__.py [hunk 3]
```diff
@@ -453,6 +453,17 @@ def munchify_cycles(obj):
         # Then finish munchifying lists and dicts inside obj (reusing munchified obj if cycles are encountered)
         return post_munchify(partial, obj)
 
+    def pre_munchify_cycles(obj):
```
Comment (NoamNol) [inline munch/__init__.py:450]: You can reuse `pre_munchify_cycles` in `munchify_cycles` (same code except the last line)

File: tests/test_munch.py (hunk 1 above)
Comment (d1618033) [reply tests/test_munch.py]: it helps for debugging to know that it's equal, but I'll remove it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (d1618033) [reply .github/workflows/ci.yml:46]: not for linting, linting can be done on just 1

File: munch/__init__.py (hunk 3 above)
Comment (d1618033) [reply munch/__init__.py:450]: refactored

File: .github/workflows/ci.yml (hunk 2 above)
Comment (NoamNol) [reply .github/workflows/ci.yml:46]: Oh right

**Pull request merged**